frontend/src/common/queries/wheresMyGene.ts

```diff
@@ -466,73 +504,266 @@ export function useGeneExpressionSummariesByTissueName(): {
+  data: TransformCellTypeGeneExpressionSummaryDataInput
 ): CellTypeGeneExpressionSummaryData {
-  const { id, pc, me, tpc, n } = data;
+  const { viewId, pc, me, tpc, n } = data;
 
   return {
     ...data,
     expressedCellCount: n,
-    id,
     meanExpression: me,
     percentage: pc,
     tissuePercentage: tpc,
+    viewId,
   };
 }
 
 interface TermIdLabels {
   cell_types: {
     [tissueID: string]: {
-      [id: string]: { name: string; depth: number; total_count: number };
+      [viewId: ViewId]: {
+        name: string;
+        order: number;
+        total_count: number;
+        viewId: ViewId;
+      };
     };
   };
   genes: { [id: string]: string };
 }
 
+/**
+ * (thuang): If BE changes the aggregated option id from "aggregated" to
+ * something else, we'll need to update it here
+ */
+export const COMPARE_OPTION_ID_FOR_AGGREGATED = "aggregated";
+
+/**
+ * (thuang): If BE changes the unknown option id from "unknown" to
+ * something else, we'll need to update it here
+ */
+export const COMPARE_OPTION_ID_FOR_UNKNOWN = "unknown";
+
+export interface CellTypeRow {
+  name: CellTypeSummary["name"];
+  order: CellTypeSummary["order"];
+  total_count: CellTypeSummary["total_count"];
+  viewId: CellTypeSummary["viewId"];
+  id: CellTypeSummary["id"];
+  optionName: string;
+  cellTypeName: string;
+}
+
 export function useTermIdLabels(): {
   data: TermIdLabels;
   isLoading: boolean;
 } {
   const requestBody = useWMGQueryRequestBody();
+
   const { data, isLoading } = useWMGQuery(requestBody);
+
+  const filterOptionNamesById = useFilterOptionNamesById();
+
   return useMemo(() => {
-    if (isLoading || !data) {
+    if (isLoading || !filterOptionNamesById || !data) {
       return {
         data: { cell_types: EMPTY_OBJECT, genes: EMPTY_OBJECT },
         isLoading,
       };
     }
+
     const {
       term_id_labels: { cell_types, genes },
     } = data;
+
     const returnCellTypes: TermIdLabels["cell_types"] = {};
-    Object.entries(cell_types).forEach(([tissueID, cell_types]) => {
-      const result: {
-        [id: string]: { name: string; depth: number; total_count: number };
-      } = {};
-      for (const {
-        cell_type_ontology_term_id,
-        cell_type,
-        depth,
-        total_count,
-      } of cell_types) {
-        result[cell_type_ontology_term_id] = {
-          depth,
-          name: cell_type,
-          total_count,
-        };
+
+    Object.entries(cell_types).forEach(
+      ([tissueID, tissueCellTypesWithCompareOptions]) => {
+        const sortedTissueCellTypesWithCompareOptions =
+          getSortedTissueCellTypesWithCompareOptions(
+            tissueCellTypesWithCompareOptions
+          );
+
+        const result: {
+          [viewId: CellTypeRow["viewId"]]: CellTypeRow;
+        } = {};
+
+        for (const cellTypeWithCompareOptions of sortedTissueCellTypesWithCompareOptions) {
+          const sortedCellTypeCompareOptions = getSortedCellTypeCompareOptions(
```

Comment:
after cellTypes are sorted, we go through each cellType to sort its compare options. Aggregate first, unknown last, and anything in between is sorted by `compareOptionId` a-z